dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/utils/Compilation.java

```diff
@@ -17,10 +17,11 @@
 	
 	public static void compileWithSmack(File file) throws Exception {
 		String name = file.getName().substring(0, file.getName().lastIndexOf('.'));
-
+		String clangFlags = System.getenv().get("CFLAGS") != null ? System.getenv().get("CFLAGS") : "";
```

Comment:
I would use `System.getenv().getOrDefault(""))` instead.
Also, it seems that there is an overload of `.getenv()` that directly takes the key in case you do not do additional checks (e.g. for `DAT3M_HOME`)